helper/certutil/helpers.go

```diff
@@ -156,50 +157,31 @@ func ParsePEMBundle(pemBundle string) (*ParsedCertBundle, error) {
 				parsedBundle.PrivateKeyBytes = pemBlock.Bytes
 			}
 		} else if certificates, err := x509.ParseCertificates(pemBlock.Bytes); err == nil {
-			switch len(certificates) {
-			case 0:
-				return nil, errutil.UserError{"pem block cannot be decoded to a private key or certificate"}
-
-			case 1:
-				if parsedBundle.Certificate != nil {
-					switch {
-					// We just found the issuing CA
-					case bytes.Equal(parsedBundle.Certificate.AuthorityKeyId, certificates[0].SubjectKeyId) && certificates[0].IsCA:
-						parsedBundle.IssuingCABytes = pemBlock.Bytes
-						parsedBundle.IssuingCA = certificates[0]
-
-					// Our saved certificate is actually the issuing CA
-					case bytes.Equal(parsedBundle.Certificate.SubjectKeyId, certificates[0].AuthorityKeyId) && parsedBundle.Certificate.IsCA:
-						parsedBundle.IssuingCA = parsedBundle.Certificate
-						parsedBundle.IssuingCABytes = parsedBundle.CertificateBytes
-						parsedBundle.CertificateBytes = pemBlock.Bytes
-						parsedBundle.Certificate = certificates[0]
-					}
-				} else {
-					switch {
-					// If this case isn't correct, the caller needs to assign
-					// the values to Certificate/CertificateBytes; assumptions
-					// made here will not be valid for all cases.
-					case certificates[0].IsCA:
-						parsedBundle.IssuingCABytes = pemBlock.Bytes
-						parsedBundle.IssuingCA = certificates[0]
-
-					default:
-						parsedBundle.CertificateBytes = pemBlock.Bytes
-						parsedBundle.Certificate = certificates[0]
-					}
-				}
-
-			default:
-				return nil, errutil.UserError{"too many certificates given; provide a maximum of two certificates in the bundle"}
-			}
+			certPath = append(certPath, &CertBlock{
```

Comment:
This seems reasonable. I was trying to code for user weirdness in the ordering of the bundle, but it's pretty well understood how it's supposed to be ordered.